### PR TITLE
feat: README updated

### PR DESCRIPTION
**Pull request by marrakonda**
- docker-compose.yaml added,
- README.md updated

related to #21

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,10 +4,92 @@
 ![unit-tests](https://github.com/goverland-labs/goverland-datasource-snapshot/workflows/unit-tests/badge.svg)
 ![golangci-lint](https://github.com/goverland-labs/goverland-datasource-snapshot/workflows/golangci-lint/badge.svg)
 
+Goverland Datasource Snapshot is a micro service designed to manage and synchronize various data sources for Goverland ecosystem.
+At the moment the focus is on Snapshot API.
+The microservice also leverages technologies such as gRPC and NATS.
+PostgresQL is used as database.
```
Comment (s-larionov) [inline README.md:10]:
```suggestion
PostgreSQL is used as database.
```

File: README.md [hunk 2]
```diff
@@ -4,10 +4,92 @@
 ![unit-tests](https://github.com/goverland-labs/goverland-datasource-snapshot/workflows/unit-tests/badge.svg)
 ![golangci-lint](https://github.com/goverland-labs/goverland-datasource-snapshot/workflows/golangci-lint/badge.svg)
 
+Goverland Datasource Snapshot is a micro service designed to manage and synchronize various data sources for Goverland ecosystem.
+At the moment the focus is on Snapshot API.
+The microservice also leverages technologies such as gRPC and NATS.
```
Comment (s-larionov) [inline README.md:9]: I think these details should be moved to other section (1-2 lines are base description, another lines are details about technoligies which were used)

File: README.md [hunk 3]
```diff
@@ -4,10 +4,92 @@
...
+PostgresQL is used as database.
+The application exposes Prometheus metrics.
+
+## Usage
+
+- The application initializes various services and workers to manage proposals, spaces, votes, and messages.
+- It sets up gRPC servers for remote procedure calls and integrates with NATS for event publishing.
```
Comment (s-larionov) [inline README.md:16]:
```suggestion
- Also it implements interface for voting on Snapshot  via gRPC calls.
```

File: README.md [hunk 4]
```diff
@@ -4,10 +4,92 @@
...
+
+## Features
+
+- Snapshot SDK Integration: facilitates interaction with Snapshot services for proposals, spaces, votes, and messages.
+- Event Publishing: utilizes NATS for event-driven architecture.
+- gRPC Services: provides gRPC-based APIs for interacting with the application.
```
Comment (s-larionov) [inline README.md:22]:
```suggestion
- gRPC Services: provides gRPC-based APIs for interacting with the Snapshot from internal services.
```

File: README.md [hunk 5]
```diff
@@ -4,10 +4,92 @@
...
+### Run Locally
+
+1. Clone the repository:
+   ```bash
+   git clone https://github.com/goverland-labs/goverland-datasource-snapshot.git
+   cd goverland-datasource-snapshot
+   ```
+2. Install the required Go dependencies:
+   ```bash
+    go mod tidy
```
Comment (s-larionov) [inline README.md:43]:
```suggestion
    go mod download
    go mod verify
```

File: README.md [hunk 6]
```diff
@@ -4,10 +4,92 @@
...
+   ```
+3. Spin up the necessary containers:
+   ```bash
+   docker-compose up -d
+   ```
+4. Run the application. Don't forget to provide your own Snapshot API key:
```
Comment (s-larionov) [inline README.md:49]: Also we should describe the .env.dist file (it's a config for running the app)

File: README.md [hunk 7]
```diff
@@ -4,10 +4,92 @@
...
+   - Run the following command to build the Docker image:
+     ```bash
+     docker build -t goverland-datasource-snapshot .
+     ```
+
+2. Run the Docker Container:
+
+   - After the image is built, you can run it as a container. Use the following command to start the container:
+     ```bash
+     docker run -d --name goverland-datasource-snapshot -p 8080:8080 goverland-datasource-snapshot
```
Comment (s-larionov) [inline README.md:80]:
```suggestion
     docker run -d --name goverland-datasource-snapshot -p 11000:11000 goverland-datasource-snapshot
```

File: docker-compose.yaml [hunk 8]
```diff
@@ -0,0 +1,12 @@
+version: '3.9'
+name: goverland-datasource-snapshot-local
+services:
+  datasource:
+    image: "postgres:11"
+    ports:
+      - "5432:5432"
```
Comment (s-larionov) [inline docker-compose.yaml:7]:
```suggestion
      - "11000:11000" # voting gRPC server
      - "3000:3000" # health check
      - "2112:2112" # prometheus metrics
```

**Review by s-larionov — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (marrakonda) [reply README.md:50]: When I run the app locally via `go run main.go`, it looks like this config is ignored. What is the right way to do it?
Btw that's why I used the following command:
   ```bash
    POSTGRES_DSN="host=localhost port=5432 user=postgres password=postgres dbname=postgres sslmode=disable" \
    SNAPSHOT_API_KEY=YOUR_KEY \
    go run main.go
   ```

File: README.md (hunk 2 above)
Comment (marrakonda) [reply README.md:9]: any ideas where?

File: README.md (hunk 6 above)
Comment (s-larionov) [reply README.md:49]: Moved to different PR

**Review by s-larionov — APPROVED** [review]

**Pull request merged**